pkg/controller/deployment/deployment_controller.go

```diff
@@ -165,20 +180,23 @@ func (dc *DeploymentController) Run(ctx context.Context, workers int) {
 	<-ctx.Done()
 }
 
-func (dc *DeploymentController) addDeployment(obj interface{}) {
+func (dc *DeploymentController) addDeployment(ctx context.Context, obj interface{}) {
+	logger := klog.FromContext(ctx)
 	d := obj.(*apps.Deployment)
-	klog.V(4).InfoS("Adding deployment", "deployment", klog.KObj(d))
+	logger.V(4).Info("Adding deployment", "deployment", klog.KObj(d))
 	dc.enqueueDeployment(d)
 }
 
-func (dc *DeploymentController) updateDeployment(old, cur interface{}) {
+func (dc *DeploymentController) updateDeployment(ctx context.Context, old, cur interface{}) {
+	logger := klog.FromContext(ctx)
```

Comment:
These functions are always going to get passed the same ctx, therefore the logger will always be the same. In such a situation it is better to look up the logger in the caller and pass a klog.Logger.